File: 백준/Silver/4134. 다음 소수/다음 소수.py
import sys,math
n=int(sys.stdin.readline())
for i in range(n):
    flag=False
    num=int(sys.stdin.readline())
    if num<2:
        print(2)
    else:
        while 1:
            if flag:
                print(num)
                break
            flag=True
            for j in range(2,int(math.sqrt(num))+1):
                if num%j==0:
                    flag=False
                    num+=1
                    break

File: 백준/Silver/4134. 다음 소수/다음 소수.c
#include <math.h>
#include <stdbool.h>
#include <stdio.h>

int main(void) {
    long long n, num;
    bool flag;
    scanf("%lld", &n);
    for (long long i = 0; i < n; i++) {
        flag = false;
        scanf("%lld", &num);
        if (num < 2) {
            printf("%d\n", 2);
        } else {
            while (1) {
                if (flag) {
                    printf("%lld\n", num);
                    break;
                }
                flag = true;
                for (long long j = 2; j < (long long)sqrt(num) + 1; j++) {
                    if (num % j == 0) {
                        flag = false;
                        num += 1;
                        break;
                    }
                }
            }
        }
    }
    return 0;
}